src/code/code_8006C510.c

```diff
@@ -1,6 +1,42 @@
 #include <ultra64.h>
 #include <global.h>
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8006C510/func_8006C510.s")
+f32 func_8006C510(f32 arg0, f32 arg1, f32 arg2, f32 arg3, f32 arg4, f32 arg5) {
+    char pad[0x1C];
+    f32 sq = SQ(arg0);
+    f32 cube = sq * arg0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8006C510/func_8006C5A8.s")
+    return (((cube + cube) - sq * 3.0f) + 1.0f) * arg2 + (sq * 3.0f - (cube + cube)) * arg3 +
+           ((cube - (sq + sq)) + arg0) * arg4 * arg1 + (cube - sq) * arg5 * arg1;
+}
+
+f32 func_8006C5A8(f32 target, TransformData* transData, s32 refIdx) {
+    s32 i;
+    s32 i1;
+
+    if (target <= transData->unk_02) {
+        return transData->unk_08;
+    }
+    if (target >= transData[refIdx - 1].unk_02) {
+        return transData[refIdx - 1].unk_08;
+    }
+
+    for (i = 0;; i++) {
+        i1 = i + 1;
```

Comment:
Have you tried just using `i + 1` where necessary instead of using a temp for it?

Reply:
i + 1 does not work.